code/prisonersDilemma.py

```diff
@@ -35,14 +17,25 @@ def strategyMove(move):
     else:
         return move
 
-def runRound(pair):
+def runRound(STRATEGY_FOLDER, pair, minGameLength=200, logMultiplier=40):
+    """
+    Returns a 2-by-n numpy array. The first axis is which player (0 = us, 1 = opponent)
+    The second axis is which turn. (0 = first turn, 1 = next turn, etc.
+    For example, it might have the values
+    
+    [[0 0 1]       a.k.a.    D D C
+     [1 1 1]]      a.k.a.    C C C
+    
+    if there have been 3 turns, and we have defected twice then cooperated once,
+    and our opponent has cooperated all three times.
+    """
     moduleA = importlib.import_module(STRATEGY_FOLDER+"."+pair[0])
     moduleB = importlib.import_module(STRATEGY_FOLDER+"."+pair[1])
     memoryA = None
     memoryB = None
     
-    LENGTH_OF_GAME = int(200-40*np.log(random.random())) # The games are a minimum of 50 turns long. The np.log here guarantees that every turn after the 50th has an equal (low) chance of being the final turn.
-    history = np.zeros((2,LENGTH_OF_GAME),dtype=int)
+    LENGTH_OF_GAME = int(minGameLength-logMultiplier*np.log(random.random())) # The games are a minimum of 50 turns long. The np.log here guarantees that every turn after the 50th has an equal (low) chance of being the final turn.

```

Comment:
The sum of `minGameLength` and `logMultiplier` is actually the average length of a game. So, if you rename `logMultiplier` to `averageGameLength` and subtract `minGameLength` from it you have even better variable names. ^^

```py
LENGTH_OF_GAME = int(minGameLength - (averageGameLength - minGameLength) * np.log(random.random()))
```

Also might wanna fix the comment while we're at it.

Reply:
I'm afraid that's what is intended by the creator or not. I will update if
thats what the creator wants.